data/transactions/logic/eval.go

```diff
@@ -685,9 +792,30 @@ func boolToUint(x bool) uint64 {
 	return 0
 }
 
-// MaxStackDepth should move to consensus params
+// MaxStackDepth should not change unless gated by a teal version change / consensus upgrade.
```

Comment:
you basically saying that it's need to be a consensus parameter..

Reply:
Not exactly.  There are many things that we gate off the  teal version. It would, in fact, be _wrong_ to make it a consensus parameter. Old programs must continue to see the value associated with their version, not a new value that appears in a new consensus parameter.  I think the old comment was misleading and implied we could change the value for all programs, new and old, by making it a consensus parameter.